packages/tsx-editor/src/components/Editor.tsx

```diff
@@ -1,12 +1,19 @@
 import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
-import React from 'react';
+import * as React from 'react';
 import { IEditorProps } from './Editor.types';
 
-export class Editor extends React.Component<IEditorProps> {
-  private editor: monaco.editor.IStandaloneCodeEditor | undefined;
-  private editorRef = React.createRef<HTMLDivElement>();
+export const Editor = (props: IEditorProps) => {
+  const ref = React.useRef<HTMLDivElement>(null);
+  const { width, height, onChange, code, language } = props;
+  const style = { width, height };
+
+  React.useEffect(() => {
```

Comment:
Still learning about hooks but I think this will cause the editor to be re-created if the editor component is re-rendered? @kenotron do you know?